force-app/main/default/classes/ContactCreationOnAccount.cls

```diff
@@ -0,0 +1,16 @@
+public with sharing class ContactCreationOnAccount {
+    
+    public static void contactCreation(List<Account> oldAccounts){
+        String contactLastName = 'DefaultContact';
+        String contactEmail = 'default@email.com';
+        List<Contact> contactToBeCreated = new List<Contact>();
+        for(Account acc : oldAccounts){
+            Contact c = new Contact();
+            c.LastName = contactLastName;
+            c.Email = contactEmail;
+            c.AccountId = acc.Id;
+            contactToBeCreated.add(c);
+        }
+        insert contactToBeCreated;
```

Comment:
This is good. The only change is that the variable name oldAccounts is confusing. I thought you were trying to work with Trigger.oldList. I had to go back to your Trigger code and check that it was Trigger.new